packages/mobile/src/exchange/ExchangeReview.tsx

```diff
@@ -172,63 +194,115 @@ export class ExchangeReview extends React.Component<Props, State> {
                     tokenName: t(`global:${this.state.inputTokenDisplayName}`),
                   })}
                 </Text>
-                <Text style={styles.currencyAmountText}>
-                  {getMoneyDisplayValue(this.state.inputAmount, this.state.inputToken, true)}
-                </Text>
+                <CurrencyDisplay
+                  style={styles.currencyAmountText}
+                  amount={{
+                    value: this.state.inputAmount.toString(),
+                    currencyCode: CURRENCIES[this.state.inputToken].code,
+                  }}
+                />
               </View>
               <HorizontalLine />
               <View style={styles.subtotalRowContainer}>
                 <Text style={styles.exchangeBodyText}>
-                  {t('subtotalAmount', {
-                    rate: getMoneyDisplayValue(exchangeRate, CURRENCY_ENUM.DOLLAR, true),
-                  })}
-                </Text>
-                <Text style={styles.exchangeBodyText}>
-                  {getMoneyDisplayValue(dollarAmount, CURRENCY_ENUM.DOLLAR, true)}
+                  <Trans i18nKey="subtotalAmount" ns={Namespaces.exchangeFlow9}>
```

Comment:
Curious on switching to the `Trans` components - when do you prefer this vs just using the `t` function?

Reply:
I tried keeping just the `t` function and passing `<CurrencyDisplay ... />` as a variable, but it always ended up being escaped to `[object Object]` not matter what I tried (`useRawValueToEscape: true`, `escapeValue: false` and their permutations).

`<Trans ... />` is the only way I found to make it work with the component approach.

One alternative would be to introduce a function returning a simple string to continue using the `t` function.
Maybe will see the need for this later on.

Reply:
Note: we would need to pass some of the redux state for currency conversion for the function approach. Which could add some boilerplate at the call site compared to the component approach.